core/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -126,6 +129,9 @@ private UnassignedInfo(Reason reason, long timestamp, String message, Throwable
     UnassignedInfo(StreamInput in) throws IOException {
         this.reason = Reason.values()[(int) in.readByte()];
         this.timestamp = in.readLong();
+        // As System.nanoTime() cannot be compared across different JVMs, reset it to now.
+        // This means that in master failover situations, elapsed delay time is forgotten.
```

Comment:
why did you decide not to do the approximation we talked about? i.e., `System.nanoTime() - (Math.min(0, System.currentTimeMillis() - this.timestap))* 1000000L;`